### PR TITLE
fix(upload): parse audio duration in JS, drop ffprobe dependency

### LINKED ISSUE #58: [Bug]: Upload Audio fails with "File does not contain a valid audio stream" - ffmpeg not installed in Docker image
## Bug Description
The "Upload Audio" feature fails for all file formats (MP3, WAV, etc.) with the error:
"File does not contain a valid audio stream"

## Root Cause
The upload route uses `ffprobe` to validate audio duration, but `ffmpeg`/`ffprobe` is not 
installed in the Docker image. This causes every upload to fail the duration check and 
return the error regardless of whether the file is valid.

## Steps to Reproduce
1. Deploy OpenPlaud using the provided docker-compose.yml
2. Go to Recordings > Upload Audio
3. Select any valid MP3 or WAV file
4. Error: "File does not contain a valid audio stream"

## Fix
Add ffmpeg to the runner stage of the Dockerfile:

RUN apt-get update -qq && apt-get install -y ffmpeg && rm -rf /var/lib/apt/lists/*

## Environment
- Debian GNU/Linux 13 (trixie) container
- Deployed via Docker Compose on TrueNAS SCALE

### Steps to Reproduce

1. Deploy OpenPlaud using the provided docker-compose.yml
2. Go to Recordings > Upload Audio
3. Select any valid MP3 or WAV file
4. See error: "File does not contain a valid audio stream"

### Expected Behavior

Audio file should upload successfully and appear in the recordings list.

### Actual Behavior

Every upload fails with "File does not contain a valid audio stream" regardless of file format (MP3, WAV, etc.)

### Deployment Type

Docker Compose (recommended)

### OpenPlaud Version

0.1.0

### Operating System

Linux

### Browser (if UI issue)

Microsoft Edge

### Logs / Error Messages

```shell
ffprobe is not installed in the Docker image. The upload route calls ffprobe to check audio duration, but the binary does not exist, causing all uploads to return duration 0 and fail validation. Fix: add "RUN apt-get update -qq && apt-get install -y ffmpeg && rm -rf /var/lib/apt/lists/*" to the runner stage of the Dockerfile.
```

### Screenshots

_No response_

### Additional Context

_No response_

### Checklist

- [x] I have searched existing issues to avoid duplicates
- [x] I have included all requested information above
- [x] I am willing to help test a fix if needed

### PR DESCRIPTION
**Pull request by kacperkwapisz**
## Summary

Replaces the `ffprobe` shell-out in the upload route with [`music-metadata`](https://github.com/Borewit/music-metadata), a pure-JS audio metadata parser. Fixes #58 for Docker users (where ffmpeg was never installed) and also fixes the same failure on local dev environments where `ffprobe` isn't on the Node process `PATH`.

## Why not just install ffmpeg in the Dockerfile?

That was #64's approach (thanks @emaxcs for the repro). It would fix Docker but leave the bug present in any local dev / CI environment without ffprobe on PATH, and would add ~70 MB of ffmpeg binaries to the runtime image purely to read a duration field. Pure JS is the modern Node convention here and removes the failure mode entirely.

## Changes

- `src/app/api/recordings/upload/route.ts` — `getAudioDurationMs` now calls `parseBuffer` from `music-metadata` against the in-memory upload buffer. Drops the temp-file write, the `node:child_process` / `node:os` / `node:util` imports, and the `tmpDir` mkdtemp/rm machinery. Parse errors are now logged via `console.error` instead of silently swallowed — the swallow was the exact reason the original bug surfaced as a misleading "invalid audio stream" response.
- `package.json` / `pnpm-lock.yaml` — adds `music-metadata@^11`.
- `src/tests/regressions/58-upload-no-ffprobe.test.ts` — regression test that asserts MP3 duration parses correctly from a Buffer with no system binary involved.
- `src/tests/fixtures/sample.mp3` — 4.5 KB / 1 s sine wave fixture.
- `CHANGELOG.md` — entry under `[Unreleased] → Fixed`.

## Format coverage

`music-metadata` covers every extension in `ACCEPTED_EXTENSIONS`: mp3, mp4, m4a, wav, ogg, opus, webm, aac, flac. The `{ duration: true }` option forces a full scan when the container header doesn't expose duration (e.g. Chrome-recorded WebM/Opus), so we don't regress those edge cases versus ffprobe.

## Testing

- `pnpm test` — all 83 tests pass, including the new regression.
- `pnpm type-check` — clean.
- `pnpm format-and-lint:fix` — clean.
- Manual: re-uploaded a real MP3 against `pnpm dev` and got a 200 with the correct duration on a host where ffprobe isn't on PATH.

Closes #58. Supersedes #64.

---
## Summary by cubic
Parse audio duration in pure JS using `music-metadata` in the upload route, removing the `ffprobe` dependency and fixing failed uploads when ffmpeg isn’t installed. Resolves #58 for Docker and any environment without `ffprobe` on PATH.

- **Bug Fixes**
  - Use `music-metadata.parseBuffer` on the in-memory upload buffer with `{ duration: true }`; works across all accepted formats.
  - Remove temp-file write and `ffprobe` shell-out; log parse errors instead of swallowing them to avoid misleading 422s.
  - Simplify the regression test to assert duration parsing from an MP3 buffer and drop brittle container/codec checks.

- **Dependencies**
  - Add `music-metadata@^11`.

<sup>Written for commit 0c4209ed645645c7fbac87a533c8207c526a5521. Summary will update on new commits.</sup>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## [Unreleased]
 
+### Fixed
+- Audio uploads no longer require `ffprobe`/`ffmpeg` on the host. Duration is now parsed in pure JS via [`music-metadata`](https://github.com/Borewit/music-metadata), fixing uploads on the published Docker image (which never shipped ffmpeg) and on local dev environments where `ffprobe` is missing from the Node process `PATH` ([#58](https://github.com/openplaud/openplaud/issues/58)).
+
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply CHANGELOG.md]: Fixed in a78fcb7 — dropped the `[Unreleased] → Fixed` entry. AGENTS.md is explicit that CHANGELOG is maintainer-curated at release time; my mistake.

File: src/tests/regressions/58-upload-no-ffprobe.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        const buffer = await readFile(FIXTURE);
+
+        const { format } = await parseBuffer(
+            buffer,
+            { mimeType: "audio/mpeg", size: buffer.byteLength },
+            { duration: true },
+        );
+
+        expect(format.container).toBe("MPEG");
+        expect(format.codec).toMatch(/MPEG \d Layer 3/);
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/tests/regressions/58-upload-no-ffprobe.test.ts]: Fixed in 0c4209e — dropped the `format.container` and `format.codec` assertions. The test now only checks `format.duration` is within tolerance, which is the actual regression surface for #58.
